### PR TITLE
DataGrid & TreeList - add return promise to navigateToRow method

### PR DESCRIPTION
**Pull request by pomahtri**
Cherry pick from https://github.com/DevExpress/DevExtreme/pull/17617

### CODE REVIEW

**Review by NickMitrokhin — DISMISSED** [review]

File: js/renovation/ui/grids/data_grid/data_grid.tsx [hunk 1]
```diff
@@ -298,7 +298,7 @@ export class DataGrid extends JSXComponent(DataGridProps) implements DataGridFor
   }
 
   @Method()
-  navigateToRow(key: any): void {
+  navigateToRow(key: any): DxPromise<number> {
```
Comment (vconst) [inline js/renovation/ui/grids/data_grid/data_grid.tsx]: Why is number?

File: js/ui/data_grid.d.ts [hunk 2]
```diff
@@ -2429,8 +2429,9 @@ export interface GridBase {
      * @param1 key:any
      * @prevFileNamespace DevExpress.ui
      * @public
+     * @return Promise<number>
      */
-    navigateToRow(key: any): void;
+    navigateToRow(key: any): DxPromise<number>;
```
Comment (vconst) [inline js/ui/data_grid.d.ts]: Why is number?

**Review by NickMitrokhin — DISMISSED** [review]

File: js/renovation/ui/grids/data_grid/data_grid.tsx [hunk 3]
```diff
@@ -291,7 +291,7 @@ export class DataGrid extends JSXComponent(DataGridProps) implements DataGridFor
   }
 
   @Method()
-  navigateToRow(key: any): void {
+  async navigateToRow(key: any): DxPromise {
```
Comment (vconst) [inline js/renovation/ui/grids/data_grid/data_grid.tsx]: Remove async because async does not work correctly now.

Comment (vconst) [reply js/renovation/ui/grids/data_grid/data_grid.tsx]: Why is not DxPromise<void> as in d.ts?

Comment (pomahtri) [reply js/renovation/ui/grids/data_grid/data_grid.tsx]: There's ts lint rule `@typescript-eslint/promise-function-async`: "Functions that return promises must be async", so I can't avoid using it as in d.ts

**Review by NickMitrokhin — APPROVED** [review]

**Review by vconst — APPROVED** [review]

**Pull request merged**